de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/git/JGitFacade.java

```diff
@@ -0,0 +1,179 @@
+package de.fu_berlin.inf.dpp.git;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.Set;
+import org.apache.log4j.Logger;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.RemoteRemoveCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.InvalidRemoteException;
+import org.eclipse.jgit.lib.NullProgressMonitor;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.transport.BundleWriter;
+import org.eclipse.jgit.transport.RefSpec;
+import org.eclipse.jgit.transport.TransportBundleStream;
+import org.eclipse.jgit.transport.URIish;
+
+public class JGitFacade {
+  private static final Logger log = Logger.getLogger(JGitFacade.class);
+
+  /**
+   * Create a bundle with all commits from basis to actual of the Git repo.
+   *
+   * @param workDir The directory that contains the .git directory
+   * @param actual The name of the ref to lookup. For example: "HEAD" or "refs/heads/master". Must
+   *     not be a short-handform; e.g., "master" is not automatically expanded
+   *     to"refs/heads/master". Include the object the ref is point at in the bundle and (if the
+   *     basis is an empty string) everything reachable from it.
+   * @param basis Assume that the recipient have at least the commit the basis is pointing to. In
+   *     order to fetch from a bundle the recipient must have the commit the basis is pointing to.
+   * @throws IllegalArgumentException workDir is null or if resolving actual leads to a not existing
+   *     ref
+   * @throws IOException failed while read/resolved parameters or while written to the bundle
+   */
+  public static byte[] createBundle(File workDir, String actual, String basis)
```

Comment:
Activities are send via XStream, but they us a client-server protocol and no peer-to-peer connection and this slow the transport down. In the expected scenario ( #324 ) I don't expect big bundles because we are talking about sending 1 commit.